calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/tool/CalendarAction.java

```diff
@@ -7701,41 +7705,14 @@ public void doDefaultview( RunData rundata, Context context )
 	 */
 	public void doPermissions(RunData data, Context context)
 	{
-		// get into helper mode with this helper tool
-		startHelper(data.getRequest(), "sakai.permissions.helper");
-
 		// setup the parameters for the helper
 		SessionState state = ((JetspeedRunData) data).getPortletSessionState(((JetspeedRunData) data).getJs_peid());
 		CalendarActionState cstate = (CalendarActionState) getState(context, data, CalendarActionState.class);
 
-		String calendarRefStr = cstate.getPrimaryCalendarReference();
-		Reference calendarRef = EntityManager.newReference(calendarRefStr);
-		String siteRef = SiteService.siteReference(calendarRef.getContext());
-
-		// setup for editing the permissions of the site for this tool, using the roles of this site, too
-		state.setAttribute(PermissionsHelper.TARGET_REF, siteRef);
+		cstate.setPrevState(cstate.getState());
+		cstate.setState(MODE_PERMISSIONS);
 
-		// ... with this description
-		state.setAttribute(PermissionsHelper.DESCRIPTION, rb.getString("java.set")
-				+ SiteService.getSiteDisplay(calendarRef.getContext()));
-
-		// ... showing only locks that are prpefixed with this
-		state.setAttribute(PermissionsHelper.PREFIX, "calendar.");
-		// ... pass the resource loader object
-		ResourceLoader pRb = new ResourceLoader("permissions");
-		HashMap<String, String> pRbValues = new HashMap<String, String>();
-		for (Iterator<Entry<String, String>> iKeys = pRb.entrySet().iterator();iKeys.hasNext();)
-		{
-			Entry entry = iKeys.next();
-			String key = (String)entry.getKey();
-			pRbValues.put(key, (String)entry.getValue());
-
-		}
-		state.setAttribute("permissionDescriptions",  pRbValues);
-		
-		String groupAware = ToolManager.getCurrentTool().getRegisteredConfig().getProperty("groupAware");
-		state.setAttribute("groupAware", groupAware != null?Boolean.valueOf(groupAware):Boolean.FALSE);
-		state.removeAttribute("menu"); //Menu not required in the permission view
+		state.setAttribute(STATE_TOOL_KEY, "calendar");
```

Comment:
I feel I'm missing something. Where is the velocity template, with the new tag in it? Did you add it?

Reply:
It's included in the velocity folder as global (velocity/tool/src/templates/permissions.vm).  Added by @bgarciaentornos : https://github.com/sakaiproject/sakai/commit/d5e3e83a1f0df77344f24f80679aaafa01a54d55

Reply:
Thanks - I forgot about that shared template.